### ISSUE TITLE
Why not .* and kron?

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaMath/TensorCore.jl#2]
I’m confused on the roll of this package: why not just use ` .* ` (instead of `  hardamard ` ) and ` kron ` (instead of ` tensor `)?

**Comment by timholy** [comment]
This package is to unicode operators `⊙` and `⊗` what IntervalSets.jl was intended for `..`. We'll hope it's easier.

https://github.com/JuliaGraphics/ColorVectorSpace.jl/issues/126

OK to close?

**Comment by dlfivefifty** [comment]
I get that, the questions was more about the functions themselves. Eg why not 
```Julia
hardamard!(dest, A, B) = broadcast!(*, dest, A, B)
```
(I guess with bounds checking to prohibit vector special cases)

**Comment by timholy** [comment]
Yep. Also more compiler-friendly. (broadcast is expensive.) But this seems a question about implementation rather than purpose, and of course we'd be happy to have improved implementations.

**Comment by dlfivefifty** [comment]
For custom arrays (like ` BandedMatrix `) broadcast will likely be faster than using indexing so a more sensible default

**Comment by timholy** [comment]
Fair enough. PR would be fine.
